packages/bpk-component-datepicker/src/themeAttributes.js

```diff
@@ -0,0 +1,27 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { themeAttributes as calendarAttributes } from 'bpk-component-calendar';
+import { themeAttributes as modalAttributes } from 'bpk-component-modal';
+import { themeAttributes as popoverAttributes } from 'bpk-component-popover';
+
+export default [
+  ...calendarAttributes,
+  ...popoverAttributes,
+  ...modalAttributes,
+].filter((attribute, index, attributes) => attributes.indexOf(attribute) === index);
```

Comment:
I'm just curious here, what this filter is supposed to catch? `attributes.indexOf(attribute) === index` will always evaluate to `true` right?

Reply:
This is a `uniq` filter. Any items that match but are not at the same index are tossed.

Reply:
In this case, modal and popover have similar (if not identical) theme attributes .. this accounts for any instance in which their theme attributes might change.

Reply:
ah! nice!